javalib/src/main/scala/java/lang/AbstractStringBuilder.scala

```diff
@@ -4,6 +4,8 @@ import java.io.InvalidObjectException
 import java.util.Arrays
 import scala.util.control.Breaks._
 
+import scala.scalanative.runtime.ieee754tostring.ryu._
+
```

Comment:
Revealed by this PR, but caused by the changes of this PR.

To separate concerns, I recommend opening an Issue
stating that this class should be `private[lang]`.
ref:  `https://www.oreilly.com/library/view/java-in-a/0596007736/re107.html`

That way, the `append0` used in this class will not leak into
the wild (anymore than it already has).

Reply:
Good point.

Actually I saw that the constructor was already `private[lang]`, so I stupidly thought that it was totally fine.
And I think things are a bit more complicated. If you look at `trimToSize()`, it is expected to be public:
https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html#trimToSize--

However, the SN StringBuilder doesn't implement an override, it is inherited from AbstractStringBuilder.
https://github.com/scala-native/scala-native/blob/main/javalib/src/main/scala/java/lang/StringBuilder.scala

Also what is sill not clear to me, is how SN actually hide all javalib classes/methods that are not part of the Java API. This magic is unclear to me.

Reply:
> Did you mean to create 2 issues about the breakable for the AbstractStringBuilder class

I believe the practice is to make them some form of `private`.  For example, I have recently 
been doing some work in `java.net`  There is a file there `SocketHelpers.scala` which is
definitely not part of the Java API.  I would have to check how the class & methods are declared.

In my own recent `java.net` work, I have needed  private constructors (to pass Array rather
that InetAddress containing Array, sound familiar?). I because of usage patterns I needed
to mark them  `private[net]`.     I would have preferred a much more limited scope, but could
not accomplish that .

Does this make sense?

Let me know if there is a particular area you want me to look at.  I do get let out of `java.net`
once in a great while. I am looking forward to Christmas parole.

Reply:
I guess that an "abstract class" having a private constructor should be totally fine and we can thus keep it like this.

However, what is most important I think, for consistency and readability, is to flag what methods are "private[lang]" (not part of Java API).
Don't you think?

Reply:
Agreed, divide & conquer, or at least have fewer concerns in the gauntlet as we try to run it.